### PR TITLE
Home task3

### PR DESCRIPTION
**Pull request by elenagronskaya**

### CODE REVIEW

File: src/common/Button/Button.jsx [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
+const Button = ({
+	buttonText,
+	showCourseButtonStyle,
+	onClick,
+	type,
+	imageLink,
+}) => (
 	<button className={showCourseButtonStyle} onClick={onClick} type={type}>
-		{buttonText}
+		{buttonText ? buttonText : <img src={imageLink} alt='button-text' />}
```
Comment (urakovaliaskar) [inline src/common/Button/Button.jsx]: `buttonText || button-text`

File: src/components/CourseInfo/CourseInfo.jsx [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
+
 	const { courseId } = useParams();
 
 	useEffect(() => {
-		setCourse(getCourseById(courseId));
+		getCourseById(courseId);
 	}, [courseId]);
 
+	useEffect(() => {
+		if (!availableAuthors || availableAuthors.length === 0) {
```
Comment (urakovaliaskar) [inline src/components/CourseInfo/CourseInfo.jsx]: !availableAuthors?.length

File: src/components/CourseInfo/CourseInfo.jsx [hunk 3]
```diff
@@ -1,19 +1,35 @@
...
+	const course = coursesData.showCourse;
+
+	const authorsData = useSelector(getAuthorsSelector);
+	const availableAuthors = authorsData?.list;
+
 	const { courseId } = useParams();
 
 	useEffect(() => {
-		setCourse(getCourseById(courseId));
+		getCourseById(courseId);
```
Comment (urakovaliaskar) [inline src/components/CourseInfo/CourseInfo.jsx]: you can remove brackets for this effect and write it in one line
`useEffect(() => getCourseById(courseId), [courseId]);`

File: src/components/CourseInfo/CourseInfo.jsx [hunk 4]
```diff
@@ -40,9 +56,13 @@ const CourseInfo = () => {
 						</p>
 						<p className={styles.infoTitle}>
 							Authors:{' '}
-							<span className={styles.infoDesc}>
-								{course?.authors.join(', ') || ''}
-							</span>
+							{course?.authors.map((authorId) => {
```
Comment (urakovaliaskar) [inline src/components/CourseInfo/CourseInfo.jsx]: course?.authors?.map

File: src/components/Courses/Courses.jsx [hunk 5]
```diff
@@ -1,38 +1,50 @@
...
+	const allCourses = coursesData?.list;
+	const searchResult = coursesData?.searchResult;
+	const availableAuthors = authorsData?.list;
 	let navigate = useNavigate();
 
 	useEffect(() => {
-		if (!isLoggedIn()) {
+		if (!userData.isAuth) {
 			navigate(ROUTE_LOGIN);
 			return;
```
Comment (urakovaliaskar) [inline src/components/Courses/Courses.jsx]: remove return

File: src/components/Courses/components/CourseCard/CourseCard.jsx [hunk 6]
```diff
@@ -15,6 +19,10 @@ const CourseCard = ({
 	id,
 }) => {
 	let navigate = useNavigate();
```
Comment (urakovaliaskar) [inline src/components/Courses/components/CourseCard/CourseCard.jsx]: use const

File: src/components/Courses/components/CourseCard/CourseCard.jsx [hunk 7]
```diff
@@ -34,11 +42,23 @@ const CourseCard = ({
...
-				/>
+				<div className={styles.buttonWrapper}>
+					<Button
+						buttonText={SHOW_COURSE}
+						showCourseButtonStyle={styles.showCourseButtonStyle}
+						onClick={() => navigate(`${ROUTE_COURSES}/${id}`)}
+					/>
+					<Button
+						imageLink={DeleteBtn}
+						onClick={() => courseDelete()}
```
Comment (urakovaliaskar) [inline src/components/Courses/components/CourseCard/CourseCard.jsx]: `onClick={courseDelete}`

File: src/store/user/reducer.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import USER_TYPES from './types';
+
+const initialState = {
+	isAuth: false,
+	name: '',
+	email: '',
+	token: '',
```
Comment (urakovaliaskar) [inline src/store/user/reducer.js]: default value - empty string or token value from localStorage.

File: src/components/Header/Header.jsx [hunk 9]
```diff
@@ -1,35 +1,35 @@
...
+import getUsersSelector from '../../store/user/selectors';
+import store from '../../store';
+import { logoutUser } from '../../store/user/actions';
 import styles from './styles.module.scss';
 
-const Header = ({ setUserName, userName }) => {
+const Header = () => {
 	let navigate = useNavigate();
 
+	const userData = useSelector(getUsersSelector);
```
Comment (urakovaliaskar) [inline src/components/Header/Header.jsx:22]: remove userName state from App

File: src/components/Courses/Courses.jsx [hunk 10]
```diff
@@ -1,38 +1,50 @@
...
+	const authorsData = useSelector(getAuthorsSelector);
 
+	const allCourses = coursesData?.list;
+	const searchResult = coursesData?.searchResult;
+	const availableAuthors = authorsData?.list;
 	let navigate = useNavigate();
 
 	useEffect(() => {
-		if (!isLoggedIn()) {
+		if (!userData.isAuth) {
```
Comment (urakovaliaskar) [inline src/components/Courses/Courses.jsx:28]: If i refresh the page, i will be navigated to login page even though i have a token stored in localStorage

**Review by urakovaliaskar — CHANGES_REQUESTED** [review]
Waiting for fixes

File: src/components/CourseInfo/CourseInfo.jsx [hunk 11]
```diff
@@ -1,18 +1,36 @@
...
+	const coursesData = useSelector(getCoursesSelector);
+	const course = coursesData.showCourse;
+
+	const authorsData = useSelector(getAuthorsSelector);
+	const availableAuthors = authorsData?.list;
+
 	const { courseId } = useParams();
 
+	const getCourseData = () => {
+		getCourseById(courseId).then();
```
Comment (AliaskarUrakovTR) [inline src/components/CourseInfo/CourseInfo.jsx]: revert to how it was

File: src/components/CourseInfo/CourseInfo.jsx [hunk 12]
```diff
@@ -1,18 +1,36 @@
...
+	};
+
+	useEffect(() => getCourseData);
```
Comment (AliaskarUrakovTR) [inline src/components/CourseInfo/CourseInfo.jsx]: Never call useEffect without 2nd argument

File: src/components/Courses/components/CourseCard/CourseCard.jsx [hunk 13]
```diff
@@ -34,11 +42,23 @@ const CourseCard = ({
...
+						onClick={() => navigate(`${ROUTE_COURSES}/${id}`)}
+					/>
+					<Button
+						imageLink={DeleteBtn}
+						onClick={courseDelete}
+						showCourseButtonStyle={styles.buttonImg}
+					/>
+					<Button
+						imageLink={UpdateBtn}
+						onClick={() => console.log('update')}
```
Comment (AliaskarUrakovTR) [inline src/components/Courses/components/CourseCard/CourseCard.jsx]: remove onClick if it does nothing

File: src/services.js [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+} from './store/courses/actions';
+import { EMAIL, TOKEN_KEY, USER_NAME } from './constants';
+import { loginError, loginSuccess } from './store/user/actions';
+import { authorListSuccess } from './store/authors/actions';
+
+export const getCourses = async () => {
+	try {
+		const response = await getAllCourses();
+
+		let data = response.data;
```
Comment (AliaskarUrakovTR) [inline src/services.js]: const { data } = response;

File: src/services.js [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+			filter === '' ||
+			item.title.toLowerCase().includes(filter.toLowerCase()) ||
+			item.id.toLowerCase().includes(filter.toLowerCase())
+	);
+	store.dispatch(searchCourseResultSuccess(courses));
+};
+
+export const getCourseById = async (id) => {
+	try {
+		let response = null;
```
Comment (AliaskarUrakovTR) [inline src/services.js]: why do you need it?

**Review by AliaskarUrakovTR — CHANGES_REQUESTED** [review]
Waiting useEffect fix

File: src/services.js [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+		response = await getCoursesId(id);
+		let data = response.data;
+		const course = data.result;
```
Comment (AliaskarUrakovTR) [inline src/services.js]: check if result exists, with if or optional chaining

**Review by urakovaliaskar — APPROVED** [review]

**Pull request merged**